### PR TITLE
fix: have ssl handshake check hostname

### LINKED ISSUE #324: Verify PSC DNS name is present as a SAN in the server certificate
Verify PSC DNS name is present as a SAN in the server certificate. It means removing the existing workaround to check the private IP address instead of the DNS name (https://github.com/GoogleCloudPlatform/alloydb-python-connector/pull/291).

### PR DESCRIPTION
**Pull request by jackwotherspoon**
For Cloud SQL we need to disable checking hostname of ssl handshake because IP address does not match SAN. However, for AlloyDB this is not the case and the IP address can be validated as the host name.

Fixes #324

### CODE REVIEW

**Comment by enocom** [comment]
We need to hold off on this until PSC DNS is populated in all existing certificates.

**Review by nancynh — APPROVED** [review]
LGTM, this is also good to merge in now that the PSC DNS is in all existing certificates

**Review by enocom — COMMENTED** [review]
This looks perfectly correct, but obviously is missing something. Thanks for all the work on this @jackwotherspoon .

**Comment by enocom** [comment]
Via @jackwotherspoon 

> "HostName" contains the fully qualified DNS hostname of the server,
   as understood by the client.  The hostname is represented as a byte
   string using ASCII encoding without a trailing dot.  

https://datatracker.ietf.org/doc/html/rfc6066#section-3

**Comment by jackwotherspoon** [comment]
TLDR;

For `wrap_socket` and passing it to SSLContext, the trailing dot is invalid. However, the hostname match requires a perfect match to the DNS entry in the SAN which has the trailing dot. Thus, here we are, stuck with the hostname match failing.

**Comment by enocom** [comment]
Going to close this for now. We'll need to revisit some server settings I think.
